From 3108faf86f61fca0510d7389f819da8588eacad7 Mon Sep 17 00:00:00 2001
From: wdlkmpx <wdlkmpx@gmail.com>
Date: Thu, 10 Feb 2022 20:11:25 -0500
Subject: [PATCH] socket-connect.c: fix segfault if ip_version == NULL

this happens in a pristine install for some reason

it didn't happen in my previous test, my environment was not suitable..

ref #144
---
 lib/socket-connect.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/lib/socket-connect.c b/lib/socket-connect.c
index 7ef66ad..16b61c5 100644
--- a/lib/socket-connect.c
+++ b/lib/socket-connect.c
@@ -109,7 +109,9 @@ static struct addrinfo * lookup_host (gftp_request *request,
   memset (&hints, 0, sizeof (hints));
   hints.ai_flags  = AI_CANONNAME;
 
-  if (strcmp(ip_version, "ipv4") == 0) {
+  if (!ip_version || !*ip_version) {
+      hints.ai_family = PF_UNSPEC;
+  } else if (strcmp(ip_version, "ipv4") == 0) {
       hints.ai_family = AF_INET;
   } else if (strcmp(ip_version, "ipv6") == 0) {
       hints.ai_family = AF_INET6;
